components/scream/src/physics/p3/p3_functions_ice_collection_impl.hpp

```diff
@@ -0,0 +1,129 @@
+#ifndef P3_FUNCTIONS_ICE_COLLECTION_IMPL_HPP
+#define P3_FUNCTIONS_ICE_COLLECTION_IMPL_HPP
+
+#include "p3_functions.hpp" // for ETI only but harmless for GPU
+
+namespace scream {
+namespace p3 {
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::ice_cldliq_collection(const Spack& rho, const Spack& temp, 
+                        const Spack& rhofaci, const Spack& f1pr04,
+                        const Spack& qitot_incld, const Spack& qc_incld, 
+                        const Spack& nitot_incld, const Spack& nc_incld, 
+                        Spack& qccol, Spack& nccol, Spack& qcshd, Spack& ncshdc)
+{
+   auto t_is_negative = temp < C::Tmelt;
+   auto qitot_incld_gt_small = qitot_incld > C::QSMALL;
+   auto qc_incld_gt_small = qc_incld > C::QSMALL;
+   constexpr auto eci = C::eci;
+ 
+   qccol.set(qitot_incld_gt_small && qc_incld_gt_small && t_is_negative, 
+             rhofaci*f1pr04*qc_incld*eci*rho*nitot_incld);
+   nccol.set(qitot_incld_gt_small && qc_incld_gt_small, 
+             rhofaci*f1pr04*nc_incld*eci*rho*nitot_incld);
+
+   // for T > 273.15, assume cloud water is collected and shed as rain drops
+   // sink for cloud water mass and number, note qcshed is source for rain mass        
+   qcshd.set(qitot_incld_gt_small && qc_incld_gt_small && !t_is_negative,  
+             rhofaci*f1pr04*qc_incld*eci*rho*nitot_incld);
+   nccol.set(qitot_incld_gt_small && qc_incld_gt_small && !t_is_negative, 
+             rhofaci*f1pr04*nc_incld*eci*rho*nitot_incld);
+   // source for rain number, assume 1 mm drops are shed
+   ncshdc.set(qitot_incld_gt_small && qc_incld_gt_small && !t_is_negative,
+              qcshd*sp(1.923e+6));
```

Comment:
This number should probably be pulled out as a constant. Ping a scientist if we need a good name for a new constant.

Reply:
Right, this is not obvious for me, so I leave it, but I added some value to be constants.

Reply:
@AaronDonahue , any idea what this constant is?

Reply:
Not off hand, but I can investigate and get back to you.

Reply:
Sorry for the delay, it's 1 over the mass of a 1mm raindrop:
`1/5.2e-7, 5.2e-7 is the mass of a 1 mm raindrop`

Reply:
It appears to be used in two places, for liq. drop shedding (here) and again for rain drop shedding.  So we should make it a constant, maybe `inv_dropmass=1.0_rtype/dropmass` where `dropmass=5.3e-7_rtype`

Reply:
For the record, I would make such constants verbose, regardless of how many times they are used. Even if you use it once, seeing `1.923e+6` is quite cryptic, where `1.0D0/dropmass` is much better.

Reply:
I agree.  It definitely makes it more clear what is going on.  And if there is one thing P3 is guilty of is hard-coded constants with no explanation.